This now mirrors the official upstream fix (not yet in a release, but should be in 1.9.3):
http://trac.nginx.org/nginx/changeset/dcae651b2a0cbd3de2f1fd5cf5b8c72627db94fd/nginx

index: src/event/ngx_event_openssl_stapling.c
===================================================================
--- a/src/event/ngx_event_openssl_stapling.c
+++ b/src/event/ngx_event_openssl_stapling.c
@@ -246,4 +246,5 @@
     staple->staple.data = buf;
     staple->staple.len = len;
+    staple->valid = NGX_MAX_TIME_T_VALUE;
 
     return NGX_OK;
